### PR TITLE
Add support for out-of-tree bindings

### PR DESCRIPTION
**Pull request by elprans**
Use the setuptools entry_point infrastructure to automatically load
binding modules installed in the environment.

Fixes: #381.

### CODE REVIEW

**Comment by pragnagopa** [comment]
cc @anthonychu 

@elprans  - Is there a way to test this before the release? publish these bits to https://test.pypi.org/?

**Comment by elprans** [comment]
@pragnagopa Added a test.

File: tests/test-binding/setup.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+##
+# Copyright (c) 2017-present MagicStack Inc.
```
Comment (pragnagopa) [inline tests/test-binding/setup.py]: magicstack?

File: azure/functions_worker/dispatcher.py [hunk 2]
```diff
@@ -64,6 +65,15 @@ def __init__(self, loop, host, port, worker_id, request_id,
         self._grpc_thread = threading.Thread(
             name='grpc-thread', target=self.__poll_grpc)
 
+    def load_bindings(self):
+        """Load out-of-tree binding implementations."""
+        services = {}
+
+        for ep in pkg_resources.iter_entry_points('azure.functions.bindings'):
+            ep.load()
```
Comment (pragnagopa) [inline azure/functions_worker/dispatcher.py]: Add system log that indicates bindings from which package are being added

File: tests/test-binding/functions/foo/function.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "scriptFile": "main.py",
+  "disabled": false,
```
Comment (pragnagopa) [inline tests/test-binding/functions/foo/function.json]: remove disabled property. Opened issue #400 to clean up other tests

File: tests/test_loader.py [hunk 4]
```diff
@@ -26,3 +31,31 @@ def test_loader_relimport(self):
...
+                func_id, r = await host.load_function('foo')
+
+            self.assertEqual(r.response.function_id, func_id)
+            self.assertEqual(r.response.result.status,
+                             protos.StatusResult.Success)
+        finally:
+            subprocess.run([
+                sys.executable, '-m', 'pip',
+                '--disable-pip-version-check',
+                'uninstall', '-y', '--quiet', 'foo-binding'
```
Comment (pragnagopa) [inline tests/test_loader.py]: installing test_binding but uninstalling foo-binding. Is this expected?

Comment (elprans) [reply tests/test_loader.py]: This is expected.

File: tests/test-binding/setup.py (hunk 1 above)
Comment (elprans) [reply tests/test-binding/setup.py]: Sorry, used the wrong boilerplate.

**Pull request merged**